add fPIC option to COMPILE_FLAGS

The static libraries in Xregion/network/rdr/rfb were linked by shared
library libvnc.so, so we should add fPIC option to COMPILE_FLAGS to fix
relocation issue.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 common/Xregion/CMakeLists.txt | 1 +
 common/network/CMakeLists.txt | 1 +
 common/rdr/CMakeLists.txt     | 1 +
 common/rfb/CMakeLists.txt     | 1 +
 4 files changed, 4 insertions(+)

diff --git a/common/Xregion/CMakeLists.txt b/common/Xregion/CMakeLists.txt
--- a/common/Xregion/CMakeLists.txt
+++ b/common/Xregion/CMakeLists.txt
@@ -3,4 +3,5 @@ add_library(Xregion STATIC
 
 if(UNIX)
   libtool_create_control_file(Xregion)
+  set_target_properties(Xregion PROPERTIES COMPILE_FLAGS -fPIC)
 endif()
diff --git a/common/network/CMakeLists.txt b/common/network/CMakeLists.txt
--- a/common/network/CMakeLists.txt
+++ b/common/network/CMakeLists.txt
@@ -5,4 +5,5 @@ add_library(network STATIC
 
 if(UNIX)
   libtool_create_control_file(network)
+  set_target_properties(network PROPERTIES COMPILE_FLAGS -fPIC)
 endif()
diff --git a/common/rdr/CMakeLists.txt b/common/rdr/CMakeLists.txt
--- a/common/rdr/CMakeLists.txt
+++ b/common/rdr/CMakeLists.txt
@@ -26,4 +26,5 @@ target_link_libraries(rdr ${RDR_LIBRARIES})
 
 if(UNIX)
   libtool_create_control_file(rdr)
+  set_target_properties(rdr PROPERTIES COMPILE_FLAGS -fPIC)
 endif()
diff --git a/common/rfb/CMakeLists.txt b/common/rfb/CMakeLists.txt
--- a/common/rfb/CMakeLists.txt
+++ b/common/rfb/CMakeLists.txt
@@ -98,4 +98,5 @@ target_link_libraries(rfb ${RFB_LIBRARIES})
 
 if(UNIX)
   libtool_create_control_file(rfb)
+  set_target_properties(rfb PROPERTIES COMPILE_FLAGS -fPIC)
 endif()
-- 
1.8.1.2

